### ISSUE TITLE
Error: Client is closed / Closing reason: Error: read ECONNRESET (data socket)

### ISSUE THREAD

**Issue by davetapley** [SamKirkland/FTP-Deploy-Action#153]
**Bug Description**

First time using the action, so not sure if this is a regression.
Followed README and action appears to work (due to #123) but log inspection shows otherwise.

Opening this issue specifically for the `Error: Client is closed` / `Closing reason: Error: read ECONNRESET (data socket)` error (see log below).

**My Action Config**
```yaml
ftp-share:
    name: Upload assets to FTP server
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
        with:
          lfs: true
          fetch-depth: 2

      - name: Update files on FTP server
       uses: SamKirkland/FTP-Deploy-Action@4.0.0
       with:
         server: ftp.<REDACTED IIS FTP SITE>.com
         username: <REDACTED>
         password: ${{ secrets.FTP_PASSWORD }}
         protocol: ftps
         security: strict
         log-level: verbose
```

**My Action Log**
```
----------------------------------------------------------------
🚀 Thanks for using ftp-deploy. Let's deploy some stuff!   
----------------------------------------------------------------
If you found this project helpful, please support it
by giving it a ⭐ on Github --> https://github.com/SamKirkland/FTP-Deploy-Action
or add a badge 🏷️ to your projects readme --> https://github.com/SamKirkland/FTP-Deploy-Action#badge
Creating local state at ./.ftp-deploy-sync-state.json
Local state created
Connected to 72.203.251.54:21 (No encryption)
< 220 Microsoft FTP Service

> AUTH TLS
< 234 AUTH command ok. Expecting TLS Negotiation.

Control socket is using: TLSv1.2
Login security: TLSv1.2
> USER <REDACTED>
< 331 Password required

> PASS ###
< 230 User logged in.

> TYPE I
< 200 Type set to I.

> STRU F
< 200 STRU F ok.

> OPTS UTF8 ON
< 200 OPTS UTF8 command successful - UTF8 encoding now ON.

> OPTS MLST type;size;modify;unique;unix.mode;unix.owner;unix.group;unix.ownername;unix.groupname;
< 501 Option not supported.

> PBSZ 0
< 200 PBSZ command successful.

> PROT P
< 200 PROT command successful.

  changing dir to ./
> MKD .
< 550 Access is denied. 

> CWD .
< 250 CWD command successful.

  dir changed
Trying to find optimal transfer strategy...
> EPSV
< 229 Entering Extended Passive Mode (|||5844|)

Optimal transfer strategy found.
> RETR .ftp-deploy-sync-state.json
----------------------------------------------------------------
No file exists on the server "./.ftp-deploy-sync-state.json" - this much be your first publish! 🎉
The first publish will take a while... but once the initial sync is done only differences are published!
If you get this message and its NOT your first publish, something is wrong.
----------------------------------------------------------------
Local Files:	973
Server Files:	973
----------------------------------------------------------------
Calculating differences between client & server
----------------------------------------------------------------
➕ Upload: favicon.ico


[SNIP]

----------------------------------------------------------------
Making changes to 973 files to sync server state
Uploading: 432 MB -- Deleting: 0 B -- Replacing: 0 B
----------------------------------------------------------------
creating folder "gmap_files/"
  changing dir to gmap_files
Error: Client is closed
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3704:29
    at new Promise (<anonymous>)
    at FTPContext.handle (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3684:16)
    at Client.sendIgnoringError (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:2745:25)
    at Client._openDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3251:20)
    at Client.ensureDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3242:24)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:1962:121
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:1823:71
    at new Promise (<anonymous>)
Closing reason: Error: read ECONNRESET (data socket)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:201:27) ***
  code: 'ECONNRESET'
***
Error: Client is closed
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3704:29
    at new Promise (<anonymous>)
    at FTPContext.handle (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3684:16)
    at Client.sendIgnoringError (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:2745:25)
    at Client._openDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3251:20)
    at Client.ensureDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3242:24)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:1962:121
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:1823:71
    at new Promise (<anonymous>)
Closing reason: Error: read ECONNRESET (data socket)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:201:27) ***
  code: 'ECONNRESET'
***
----------------------------------------------------------------
Time spent hashing:               1.7 seconds
Time spent connecting to server:  902 milliseconds
Time spent deploying:             3 milliseconds (156 GB/second)
  - changing dirs:                126 milliseconds
  - logging:                      28 milliseconds
----------------------------------------------------------------
Total time:                       3 seconds
----------------------------------------------------------------

```

**IIS Log**
```
2020-12-31 20:11:38 52.184.187.148 - <REDACTED> 21 ControlChannelOpened - - 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:38 52.184.187.148 - <REDACTED> 21 AUTH TLS 234 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:38 52.184.187.148 - <REDACTED> 21 USER <REDACTED> 331 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:38 52.184.187.148 <REDACTED> <REDACTED> 21 PASS *** 230 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 /
2020-12-31 20:11:38 52.184.187.148 <REDACTED> <REDACTED> 21 TYPE I 200 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:38 52.184.187.148 <REDACTED> <REDACTED> 21 STRU F 200 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:38 52.184.187.148 <REDACTED> <REDACTED> 21 OPTS UTF8+ON 200 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 OPTS MLST+type;size;modify;unique;unix.mode;unix.owner;unix.group;unix.ownername;unix.groupname; 501 87 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 PBSZ 0 200 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 PROT P 200 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 MKD . 550 5 5 a1147643-fb62-4d6f-a4ce-97843f640354 /
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 CWD . 250 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 /
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 EPSV - 229 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 5822 DataChannelOpened - - 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 5822 DataChannelClosed - - 2 3 a1147643-fb62-4d6f-a4ce-97843f640354 -
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 RETR .ftp-deploy-sync-state.json 550 2 3 a1147643-fb62-4d6f-a4ce-97843f640354 /.ftp-deploy-sync-state.json
2020-12-31 20:11:39 52.184.187.148 <REDACTED> <REDACTED> 21 ControlChannelClosed - - 0 0 a1147643-fb62-4d6f-a4ce-97843f640354 -
```

**Comment by davetapley** [comment]
# Work around

I was able to get it to work on `3.1.1` with ⬇️, although I then ran in to #114 

```
    steps:
      - uses: actions/checkout@v2
        with:
          lfs: true
          fetch-depth: 2

      - name: Update files on FTP server
        uses: SamKirkland/FTP-Deploy-Action@3.1.1
        with:
          ftp-server: ftpes://ftp.<REDACTED>.com
          ftp-username: <REDACTED>
          ftp-password: ${{ secrets.FTP_PASSWORD }}
```

**Comment by superhize** [comment]
Wanted to make a new issue, but since i have exactly the same problem, i will post here.

Will try https://github.com/SamKirkland/FTP-Deploy-Action/issues/153#issuecomment-753198797 and update my post

main.yml
```yaml
name: "🚀 Upload to FTP"
on:
  push:
    branches:
    - main
jobs:
  web-deploy:
    name: 🎉 Deploy
    runs-on: ubuntu-latest
    steps:
    - name: 🚚 Get latest code
      uses: actions/checkout@v2.3.2
      with:
          lfs: true
          fetch-depth: 2
    - name: 📂 Sync files
      uses: SamKirkland/FTP-Deploy-Action@4.1.0
      with:
        server: ${{ secrets.FTP_HOST }}
        username: ${{ secrets.FTP_USER }}
        password: ${{ secrets.FTP_PASSWORD }}
        exclude: "[**/.git*/**, **/node_modules/**, **/build/, .README.MD, .gitignore]"
        protocol: ftps
        log-level: verbose
```


action log:
```
Run SamKirkland/FTP-Deploy-Action@4.1.0
  with:
    server: ***
    username: ***
    password: ***
    exclude: [**/.git*/**, **/node_modules/**, **/build/, .README.MD, .gitignore]
    protocol: ftps
    log-level: verbose
----------------------------------------------------------------
🚀 Thanks for using ftp-deploy. Let's deploy some stuff!   
----------------------------------------------------------------
If you found this project helpful, please support it
by giving it a ⭐ on Github --> https://github.com/SamKirkland/FTP-Deploy-Action
or add a badge 🏷️ to your projects readme --> https://github.com/SamKirkland/FTP-Deploy-Action#badge
Using the following excludes filters: ["**/.git*/**","**/node_modules/**","**/build/",".README.MD",".gitignore"]
Creating local state at ./.ftp-deploy-sync-state.json
Local state created
Connected to ***:21 (No encryption)
< 220 Multicraft 2.3.5 FTP server

> AUTH TLS
< 234 AUTH TLS successful.

Control socket is using: TLSv1.3
Login security: TLSv1.3
> USER ***
< 331 Username ok, send password.

> PASS ###
< 230 Login successful

> FEAT
< 211-Features supported:

<  AUTH SSL
 AUTH TLS
 EPRT
 EPSV
 MDTM
 MFMT
 MLST type*;perm*;size*;modify*;unique*;unix.mode;unix.uid;unix.gid;
 PBSZ
 PROT
 REST STREAM
 SIZE
 TVFS
 UTF8

< 211 End FEAT.

> TYPE I
< 200 Type set to: Binary.

> STRU F
< 200 File transfer structure set to: F.

> OPTS UTF8 ON
< 200 OK

> OPTS MLST type;size;modify;unique;unix.mode;unix.owner;unix.group;unix.ownername;unix.groupname;
< 200 MLST OPTS type;size;modify;unique;unix.mode;

> PBSZ 0
< 200 PBSZ=0 successful.

> PROT P
< 200 Protection set to Private

  changing dir to ./
> MKD .
< 550 Unknown error 198.

> CWD .
< 250 "/" is the current directory.

  dir changed
Trying to find optimal transfer mode...
> EPSV
< 229 Entering extended passive mode (|||54117|).

Optimal transfer mode found.
> RETR .ftp-deploy-sync-state.json
----------------------------------------------------------------
No file exists on the server "./.ftp-deploy-sync-state.json" - this must be your first publish! 🎉
The first publish will take a while... but once the initial sync is done only differences are published!
If you get this message and its NOT your first publish, something is wrong.
----------------------------------------------------------------
Local Files:	90
Server Files:	90
----------------------------------------------------------------
Calculating differences between client & server
----------------------------------------------------------------
📁 [...]
----------------------------------------------------------------
Making changes to 90 files/folders to sync server state
Uploading: 31 kB -- Deleting: 0 B -- Replacing: 0 B
----------------------------------------------------------------
creating folder "Banker/"
  changing dir to Banker

----------------------------------------------------------------
--------------  🔥🔥🔥 an error occurred  🔥🔥🔥  --------------
----------------------------------------------------------------

----------------------------------------------------------------
----------------------  full error below  ----------------------
----------------------------------------------------------------

Error: Client is closed
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:3890:29
    at new Promise (<anonymous>)
    at FTPContext.handle (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:3870:16)
    at Client.sendIgnoringError (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:2918:25)
    at Client._openDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:3436:20)
    at Client.ensureDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:3427:24)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:2256:121
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.1.0/dist/index.js:2241:71
    at new Promise (<anonymous>)
Closing reason: Error: read ECONNRESET (data socket)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:201:27) {
  code: 'ECONNRESET'
}
Error: Error: Client is closed
```

**Comment by superhize** [comment]
**### TLDR if you don't want to read everything:**
I guess, maybe, Action is trying to connect with Implicit TLS/SSL while i need Explicit TLS/SSL -> how can i force Action to use Explicit ?





So, i've been looking at it for hours now, can't get it to work with 3.1.1


To be clear, i'm trying to upload a whole repository to a minecraft plugin folder. I'm using Multicraft built-in ftp server. FTPS connection work well with winscp with these settings:

![RealisticCoati749](https://user-images.githubusercontent.com/17607478/124370973-4dd8ab00-dc7d-11eb-998d-9dffe7abaa37.png)


For some reason, i can't connect  to ftps://user@host:21 in my brower, it open WinSCP and i got this:
```
2021-07-04 03:42:29.343 --------------------------------------------------------------------------
. 2021-07-04 03:42:29.343 Session name: user@host (Ad-Hoc site)
. 2021-07-04 03:42:29.343 Host name: host (Port: 21)
. 2021-07-04 03:42:29.343 User name: user (Password: Yes, Key file: No, Passphrase: No)
. 2021-07-04 03:42:29.343 Transfer Protocol: FTP
. 2021-07-04 03:42:29.343 Ping type: Dummy, Ping interval: 30 sec; Timeout: 15 sec
. 2021-07-04 03:42:29.343 Disable Nagle: No
. 2021-07-04 03:42:29.343 Proxy: None
. 2021-07-04 03:42:29.343 Send buffer: 262144
. 2021-07-04 03:42:29.343 UTF: Auto
**. 2021-07-04 03:42:29.343 FTPS: Implicit TLS/SSL [Client certificate: No]**
. 2021-07-04 03:42:29.343 FTP: Passive: Yes [Force IP: Auto]; MLSD: Auto [List all: Auto]; HOST: Auto
. 2021-07-04 03:42:29.343 Session reuse: Yes
. 2021-07-04 03:42:29.343 TLS/SSL versions: TLSv1.0-TLSv1.3
. 2021-07-04 03:42:29.343 Local directory: default, Remote directory: home, Update: Yes, Cache: Yes
. 2021-07-04 03:42:29.343 Cache directory changes: Yes, Permanent: Yes
. 2021-07-04 03:42:29.343 Recycle bin: Delete to: No, Overwritten to: No, Bin path: 
. 2021-07-04 03:42:29.343 Timezone offset: 0h 0m
. 2021-07-04 03:42:29.343 --------------------------------------------------------------------------
. 2021-07-04 03:42:29.639 Connecting to host ...
. 2021-07-04 03:42:29.639 Connected with host, negotiating TLS connection...
. 2021-07-04 03:42:29.639 TLS connect: error in error
. 2021-07-04 03:42:29.639 Can't establish TLS connection
. 2021-07-04 03:42:29.639 Disconnected from server
. 2021-07-04 03:42:29.639 Connection failed.
* 2021-07-04 03:42:29.692 (EFatal) Connection failed.
* 2021-07-04 03:42:29.692 TLS connect: error in error
* 2021-07-04 03:42:29.692 Can't establish TLS connection
* 2021-07-04 03:42:29.692 Disconnected from server
* 2021-07-04 03:42:29.692 Connection failed.*
```


What i'm seeing here is he's trying to connect with **Implicit TLS/SSL**, wich is not what i want, i'm guessing Action does the same thing, but how can i change it to **Excplicit TLS/SSL** 


action config
```yaml
name: "🚀 Upload to FTP"
on:
  push:
    branches:
    - main
jobs:
  web-deploy:
    name: 🎉 Deploy
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2.1.0
        with:
          fetch-depth: 2
      - name: Update files on FTP server
        uses: SamKirkland/FTP-Deploy-Action@3.1.1
        with:
          ftp-server: ftps://...:21
          ftp-username: ${{ secrets.FTP_USER }}
          ftp-password: ${{ secrets.FTP_PASSWORD }}
          git-ftp-args: --insecure
```

action log:
```
Run SamKirkland/FTP-Deploy-Action@3.1.1
  with:
    ftp-server: ftps://...:21
    ftp-username: ***
    ftp-password: ***
    git-ftp-args: --insecure
    local-dir: ./
/usr/bin/docker run --name a33c1039698323d7c431e8af61519469ba237_ebf1c4 --label 8a33c1 --workdir /github/workspace --rm -e INPUT_FTP-SERVER -e INPUT_FTP-USERNAME -e INPUT_FTP*** INPUT_GIT-FTP-ARGS -e INPUT_LOCAL-DIR -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/irezia_betonquest/irezia_betonquest":"/github/workspace" 8a33c1:039698323d7c431e8af61519469ba237
Uploading files
  /usr/bin/git ftp push --force --auto-init --verbose --syncroot=./ --user=*** --passwd=*** --insecure ftps://...:21
  Sun Jul  4 01:52:41 UTC 2021: git-ftp version 1.5.1 running on Linux a4db21477e6f 5.8.0-1036-azure #38~20.04.1-Ubuntu SMP Thu Jun 17 14:14:18 UTC 2021 x86_64 GNU/Linux
  Sun Jul  4 01:52:41 UTC 2021: Forced mode enabled.
  Sun Jul  4 01:52:41 UTC 2021: Auto init if needed.
  Sun Jul  4 01:52:41 UTC 2021: Using syncroot ./ if exists.
  Sun Jul  4 01:52:41 UTC 2021: Insecure SSL/TLS connection allowed
  Sun Jul  4 01:52:41 UTC 2021: Host is '...:21'.
  Sun Jul  4 01:52:41 UTC 2021: User is '***'.
  Sun Jul  4 01:52:41 UTC 2021: Password is set.
  Sun Jul  4 01:52:41 UTC 2021: CACert is ''.
  Sun Jul  4 01:52:41 UTC 2021: Insecure is '1'.
  Sun Jul  4 01:52:41 UTC 2021: Proxy is ''.
  Sun Jul  4 01:52:41 UTC 2021: Path is ''.
  Sun Jul  4 01:52:41 UTC 2021: Syncroot is './'.
  Sun Jul  4 01:52:41 UTC 2021: The remote sha1 is saved in file '.git-ftp.log'.
  Sun Jul  4 01:52:41 UTC 2021: Check if curl is functional.
  Sun Jul  4 01:52:41 UTC 2021: Retrieving last commit from ***...:21/.
  #=#=#                                                                         
  ##O#- #                                                                       
  curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number
  Sun Jul  4 01:52:42 UTC 2021: Check if ***...:21 is accessible.
  #=#=#                                                                         
  ##O#- #                                                                       
  curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number
  Sun Jul  4 01:52:42 UTC 2021: fatal: Can't access remote '***...:21', exiting...
Error: The process '/usr/bin/git' failed with exit code 5
✅ Deploy Complete
```

**Comment by superhize** [comment]
https://github.com/SamKirkland/FTP-Deploy-Action/issues/153#issuecomment-753198797 is working after hours. Only cause i thought ftpes:// was a mistake so i replaced it with ftps://


God damn i'm stupid.

**Comment by ronny-springer** [comment]
I'm struggling to create a FTP with explizit TLS connect.

If I understand the code correct, the [`protocol` should be FTPS](https://github.com/SamKirkland/ftp-deploy/blob/198e42578e83114191d3ed2377cce76c66d3abec/src/deploy.ts#L34C5-L36C6) to [pass down the `secrue = true`](https://www.npmjs.com/package/basic-ftp#user-content-usage). So, my YML looks like:
```      
    - name: Sync files
        uses: SamKirkland/FTP-Deploy-Action@v4.3.4
        with:
          server: HOST
          protocol: ftps
          port: 21
          username: ${{ secrets.username }}
          password: ${{ secrets.password }}
          local-dir: ./build/
          server-dir: ./html/
          dry-run: false
          log-level: verbose
```

The pipeline action prints an error:
```
Error: Client is closed because read ECONNRESET (data socket)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:5197:29
    at new Promise (<anonymous>)
    at FTPContext.handle (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:5[179](https://github.com/tsv-merschwitz/website/actions/runs/5485636413/jobs/9994685677#step:5:180):16)
    at Client.sendIgnoringError (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:4226:25)
    at Client._openDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:4744:20)
    at Client.ensureDir (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:4735:24)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:3555:126
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:3540:71
    at new Promise (<anonymous>)
Closing reason: Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
  code: 'ECONNRESET'
}
Error: Error: Client is closed because read ECONNRESET (data socket)
```

How can I establish a ftp connection to publish my files?
Protokoll: FTP
Verschlüsselung: FTPS (explizites FTP über TLS)
Port: 21

I checked the connection with WinSCP. It's working.

**Comment by emyb** [comment]
Lies, I think I just got lucky. Still not happening on large uploads.

**Comment by emyb** [comment]
The workaround for adding an empty sync state json file seems to work. This is the second time this workaround has been helpful.

**Comment by tbmsu** [comment]
> The workaround for adding an empty sync state json file seems to work. This is the second time this workaround has been helpful.

Worked for me as well 😄

**Comment by neilzuk** [comment]
> > The workaround for adding an empty sync state json file seems to work. This is the second time this workaround has been helpful.
> 
> Worked for me as well 😄

I have also ran into this problem for the initial deployment and adding the empty `.ftp-deploy-sync-state.json` file workaround did the trick!

**Comment by chemseddine201** [comment]
same solution for me, i just create an empty ".ftp-deploy-sync-state.json" file on my repo and the problem fixed.

**Comment by dennisrahmen** [comment]
Always coming back to this issue when after months I need to create a new deployment action and already forgotten that I need that empty file.
Any updates on a fix?

**Comment by Anjali-coder-design** [comment]
![IMG20240614123706](https://github.com/SamKirkland/FTP-Deploy-Action/assets/66879222/b99ea164-be5d-4efb-95df-06fe45768a51)
I am facing this error while deploying to synapse workspace using GitHub actions. Please help to understand why this issue is coming and how to fix it.

**Comment by awaismukhtar92** [comment]
did it for me too lol
